pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -306,21 +331,37 @@ func (is *InstanceService) InstanceCreate(clusterName string, name string, confi
 					networkID: netID,
 				})
 			}
-		} else if net.UUID == "" && net.SubnetID != "" {
-			subnet, err := subnets.Get(is.networkClient, net.SubnetID).Extract()
-			if err != nil {
-				return nil, err
+		} else if net.Subnets != nil {
```

Comment:
This could become an else, or might go away. When net.Subnets are empty, the for loop just don't kick in.

Reply:
I think we should keep this one, tho.

This prevents from adding the same network twice, one with a subnet and one by itself. For example:

```
networks:
- uuid: MYUUID
  subnets:
    - name: test
```

Without this clause, the above example could end up with 2 `ServerNetwork` objects. One obtained by filtering the subnet and another one because `uuid` is set.

Reply:
Actually, now that I read this again, I think this code should be re-organized.